pkg/reconciler/support_archive_controller.go

```diff
@@ -30,7 +38,42 @@ func (s *SupportArchiveReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	logger.Info(fmt.Sprintf("Reconciler is triggered by resource %q", req.NamespacedName))
 
-	return ctrl.Result{}, nil
+	archiveInterface := s.client.SupportArchives(req.Namespace)
+	cr, err := archiveInterface.Get(ctx, req.Name, metav1.GetOptions{})
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !slices.Contains(cr.GetFinalizers(), finalizerName) {
+		logger.Info(fmt.Sprintf("Adding finalizer to support archive %q", cr.Name))
+		cr.Finalizers = append(cr.Finalizers, finalizerName)
+		_, updateErr := archiveInterface.Update(ctx, cr, metav1.UpdateOptions{})
+		return ctrl.Result{Requeue: true}, updateErr
+	}
+
+	if !cr.GetDeletionTimestamp().IsZero() {
+		cleanupErr := s.cleaner.Clean(ctx, cr.Name, cr.Namespace)
+		if cleanupErr != nil {
+			// Do not return here to avoid blocking in error case.
+			// Garbage collection can try to clean up inconsistent files later.
+			logger.Info(fmt.Sprintf("Failed to clean up for support archive request %q: %v", cr.Name, cleanupErr))
+		}
+
+		_, updateErr := archiveInterface.RemoveFinalizer(ctx, cr, finalizerName)
+		if updateErr != nil {
+			return ctrl.Result{Requeue: true}, updateErr
+		}
+
+		return ctrl.Result{Requeue: false}, nil
+	}
+
+	// Do not recreate archives
+	if cr.Status.Phase == libv1.StatusPhaseCreated {
+		return ctrl.Result{}, nil
+	}
```

Comment:
Is the phase still needed with conditions? Also, I thought, depending on the status for reconciliation is not best practice, right? So is this really necessary?